### PR TITLE
fix: redirect logged user to correct route

### LINKED ISSUE #6358: Soft migration fails due to 403 from api.avax.network
## Problem Definition

A user reported on Discord that they kept seeing the following error (check Traceback)

The error does not block the login to the app, but because of the error the data migration never completes and it is always retried.

## Logs

```python
29/06/2023 18:23:06 CEST] ERROR rotkehlchen.data_migrations.manager Greenlet-5: Failed to run soft data migration to version 10 due to 403 Client Error: Forbidden for url: https://api.avax.network/ext/bc/C/rpc
Traceback (most recent call last):
  File "rotkehlchen/data_migrations/manager.py", line 73, in _perform_migration
  File "rotkehlchen/data_migrations/migrations/migration_10.py", line 59, in data_migration_10
  File "rotkehlchen/chain/aggregator.py", line 1459, in detect_evm_accounts
  File "rotkehlchen/chain/aggregator.py", line 1382, in check_chains_and_add_accounts
  File "rotkehlchen/chain/aggregator.py", line 1323, in check_single_address_activity
  File "web3/module.py", line 57, in caller
  File "web3/manager.py", line 197, in request_blocking
  File "web3/manager.py", line 150, in _make_request
  File "web3/middleware/formatting.py", line 94, in middleware
  File "web3/middleware/gas_price_strategy.py", line 90, in middleware
  File "web3/middleware/formatting.py", line 94, in middleware
  File "web3/middleware/attrdict.py", line 33, in middleware
  File "web3/middleware/formatting.py", line 94, in middleware
  File "web3/middleware/formatting.py", line 94, in middleware
  File "web3/middleware/formatting.py", line 94, in middleware
  File "web3/middleware/buffered_gas_estimate.py", line 40, in middleware
  File "web3/middleware/exception_retry_request.py", line 105, in middleware
  File "web3/providers/rpc.py", line 88, in make_request
  File "web3/_utils/request.py", line 113, in make_post_request
  File "requests/models.py", line 1021, in raise_for_status
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.avax.network/ext/bc/C/rpc
```

### System Description

Operating system: 
Rotki version: 1.29

### PR DESCRIPTION
**Pull request by lukicenturi**
## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

This fix the issue with restarting on vite optimization, hanging on white screen (when I already logged in, but still on the login page and wasn't managed to be redirected to dashboard, because the restart happen before that)

### CODE REVIEW

**Review by kelsos — COMMENTED** [review]
@tewshi can you check if this works as expected and merge if it does?

File: frontend/app/src/router/index.ts [hunk 1]
```diff
@@ -466,7 +466,7 @@ router.beforeEach((to, from, next) => {
     if (
       [Routes.USER, Routes.USER_CREATE, Routes.USER_LOGIN].includes(to.path)
     ) {
-      return;
+      next(Routes.DASHBOARD);
     }
     next();
```
Comment (tewshi) [inline frontend/app/src/router/index.ts:471]: @lukicenturi will this not mean that `next` will be called twice?

so, either `return next(Routes.DASHBOARD)` or put `next()` in an else.

Comment (lukicenturi) [reply frontend/app/src/router/index.ts]: Updated

**Comment by lukicenturi** [comment]
btw, if you want to try locally,
checkout this PR, and go to `vite.config.ts` and remove the `optimizeDeps` part.

then do `pnpm run clean:modules && pnpm install && pnpm run dev`
This way, it will trigger the optimization again
```
rotki 3:48:27 PM [vite] ✨ new dependencies optimized: lodash/keys, lodash/pick, imask
3:48:27 PM [vite] ✨ optimized dependencies changed. reloading
```

when it's happening, see the app if it goes to white screen, then after several seconds, it should go to dashboard, instead of hanging on the white screen.

**Comment by tewshi** [comment]
> btw, if you want to try locally, checkout this PR, and go to `vite.config.ts` and remove the `optimizeDeps` part.
> 
> then do `pnpm run clean:modules && pnpm install && pnpm run dev` This way, it will trigger the optimization again
> 
> ```
> rotki 3:48:27 PM [vite] ✨ new dependencies optimized: lodash/keys, lodash/pick, imask
> 3:48:27 PM [vite] ✨ optimized dependencies changed. reloading
> ```
> 
> when it's happening, see the app if it goes to white screen, then after several seconds, it should go to dashboard, instead of hanging on the white screen.

looks good at my end, but check the failing pipeline.

**Comment by kelsos** [comment]
@lukicenturi @tewshi all good with this? can we merge?

File: docs/changelog.rst [hunk 2]
```diff
@@ -2,6 +2,7 @@
 Changelog
 =========
 
+* :bug:`` Fix issue where users cannot add non EVM asset.
```
Comment (kelsos) [inline docs/changelog.rst:5]:
```suggestion
* :bug:`-` Fix issue where users cannot add non-EVM assets.
```

We add a `-` when there is no issue number.

**Pull request merged**